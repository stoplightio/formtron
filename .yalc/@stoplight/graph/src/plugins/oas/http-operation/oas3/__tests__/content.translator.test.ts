import { translateHeaderObject, translateMediaTypeObject } from '../helpers/content.translator';

describe('translateMediaTypeObject', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  test('given single example should translate to IHttpContent', () => {
    expect(
      translateMediaTypeObject(
        {
          schema: {},
          example: { summary: 'single example' },
          encoding: {},
        },
        'mediaType'
      )
    ).toMatchSnapshot();
  });

  test('given multiple examples should translate to IHttpContent', () => {
    expect(
      translateMediaTypeObject(
        {
          schema: {},
          examples: [{ summary: 'multi example' }],
          encoding: {},
        },
        'mediaType'
      )
    ).toMatchSnapshot();
  });

  test('given encodings should translate each encoding to array item', () => {
    expect(
      translateMediaTypeObject(
        {
          schema: {},
          examples: [{ summary: 'multi example' }],
          encoding: {
            enc1: {
              contentType: 'text/plain',
              headers: {
                h1: {},
              },
              style: 'form',
              explode: true,
              allowReserved: true,
            },
            enc2: {
              contentType: 'text/plain',
              headers: {
                h1: {},
              },
              style: 'form',
              explode: true,
              allowReserved: true,
            },
          },
        },
        'mediaType'
      )
    ).toMatchSnapshot();
  });

  test('given complex nested media type object should translate correctly', () => {
    expect(
      translateMediaTypeObject(
        {
          schema: {},
          examples: [{ summary: 'multi example' }],
          encoding: {
            enc1: {
              contentType: 'text/plain',
              style: 'form',
              headers: {
                h1: {
                  description: 'descr',
                  style: 'matrix',
                  examples: {
                    a: {
                      summary: 'example summary',
                    },
                  },
                  content: {
                    mediaType: 'nested/media',
                  },
                },
              },
            },
          },
        },
        'mediaType'
      )
    ).toMatchSnapshot();
  });

  test('given encoding with incorrect style should throw an error', () => {
    const testedFunction = () => {
      translateMediaTypeObject(
        {
          schema: {},
          examples: [{ summary: 'multi example' }],
          encoding: {
            enc1: {
              contentType: 'text/plain',
              style: 'xyz',
            },
          },
        },
        'mediaType'
      );
    };
    expect(testedFunction).toThrowErrorMatchingSnapshot();
  });
});

describe('translateHeaderObject', () => {
  test('should translate to IHttpHeaderParam', () => {
    expect(
      translateHeaderObject(
        {
          description: 'descr',
          required: true,
          deprecated: true,
          allowEmptyValue: true,
          style: 'matrix',
          explode: true,
          allowReserved: true,
          schema: {},
          examples: {
            a: {
              summary: 'example summary',
            },
          },
          example: {},
          content: {},
        },
        'header-name'
      )
    ).toMatchSnapshot();
  });
});
