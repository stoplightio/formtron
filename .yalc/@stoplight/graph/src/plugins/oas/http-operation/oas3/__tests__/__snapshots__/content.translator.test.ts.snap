// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translateHeaderObject should translate to IHttpHeaderParam 1`] = `
Object {
  "allowEmptyValue": true,
  "allowReserved": true,
  "content": Object {},
  "deprecated": true,
  "description": "descr",
  "explode": true,
  "name": "header-name",
  "required": true,
  "schema": Object {},
  "style": "simple",
}
`;

exports[`translateMediaTypeObject given complex nested media type object should translate correctly 1`] = `
Object {
  "encodings": Array [
    Object {
      "contentType": "text/plain",
      "headers": Array [
        Object {
          "content": Object {
            "mediaType": Object {
              "encodings": Array [],
              "examples": Array [],
              "mediaType": "mediaType",
              "schema": undefined,
            },
          },
          "description": "descr",
          "name": "h1",
          "style": "simple",
        },
      ],
      "mediaType": "text/plain",
      "property": "enc1",
      "style": "form",
    },
  ],
  "examples": Array [
    Object {
      "summary": "multi example",
    },
  ],
  "mediaType": "mediaType",
  "schema": Object {},
}
`;

exports[`translateMediaTypeObject given encoding with incorrect style should throw an error 1`] = `"Encoding property style: 'xyz' is incorrect, must be one of: form,spaceDelimited,pipeDelimited,deepObject"`;

exports[`translateMediaTypeObject given encodings should translate each encoding to array item 1`] = `
Object {
  "encodings": Array [
    Object {
      "allowReserved": true,
      "contentType": "text/plain",
      "explode": true,
      "headers": Array [
        Object {
          "content": Object {},
          "name": "h1",
          "style": "simple",
        },
      ],
      "mediaType": "text/plain",
      "property": "enc1",
      "style": "form",
    },
    Object {
      "allowReserved": true,
      "contentType": "text/plain",
      "explode": true,
      "headers": Array [
        Object {
          "content": Object {},
          "name": "h1",
          "style": "simple",
        },
      ],
      "mediaType": "text/plain",
      "property": "enc2",
      "style": "form",
    },
  ],
  "examples": Array [
    Object {
      "summary": "multi example",
    },
  ],
  "mediaType": "mediaType",
  "schema": Object {},
}
`;

exports[`translateMediaTypeObject given multiple examples should translate to IHttpContent 1`] = `
Object {
  "encodings": Array [],
  "examples": Array [
    Object {
      "summary": "multi example",
    },
  ],
  "mediaType": "mediaType",
  "schema": Object {},
}
`;

exports[`translateMediaTypeObject given single example should translate to IHttpContent 1`] = `
Object {
  "encodings": Array [],
  "examples": Array [
    Object {
      "summary": "single example",
    },
  ],
  "mediaType": "mediaType",
  "schema": Object {},
}
`;
